input/otlp/metadata_test.go

```diff
@@ -335,13 +337,148 @@ func TestResourceLabels(t *testing.T) {
 		"int_array":    []interface{}{123, 456},
 	})
 	assert.Equal(t, modelpb.Labels{
-		"string_array": {Global: true, Values: []string{"abc", "def"}},
+		"string_array": {
+			Global: true,
+			Values: []string{"abc", "def"},
+		},
 	}, modelpb.Labels(metadata.Labels))
 	assert.Equal(t, modelpb.NumericLabels{
-		"int_array": {Global: true, Values: []float64{123, 456}},
+		"int_array": {
+			Global: true,
+			Values: []float64{123, 456},
+		},
 	}, modelpb.NumericLabels(metadata.NumericLabels))
 }
 
+func TestTranslateSpanValue(t *testing.T) {
```

Comment:
As discussed, this can be merged into `TestTranslateSpanSlice`, and we should also test TranslateTransaction.

Reply:
Testing for `TranslateTransaction` is a good idea. Thanks.

However, I do not see the value in merging `TestTranslateSpanValue` and `TestTranslateSpanSlice`:

1. It'll add some redundant logic to the Table Driven Test.
2. Fundamentally, a value and slice are two different data structures.
    - A value is atomic, while an array is divisible.
    - I generally like to "split" tests based around the data structure under test (if possible).
3. The `TestTranslateSpanSlice` is already lengthy and might still expand in the future if I missed edge-cases.

I'm open to follow your advice. I just want to have these conversations explicitly, so I can upgrade my thinking.

Reply:
After having a deeper look, it seems to me that `TestArrayLabels` actually already tests the underlying functions `TranslateTransaction` and `TranslateSpan`. I extended this test to ensure `setLabel` properly drops the heterogeneous elements.

Same goes for `TestResourceLabels`. I do not know how I missed this in the beginning. 🤦